### PR TITLE
Upgrade the versions of astroid and dill

### PR DESCRIPTION
**Pull request by Pierre-Sassoulas**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |

## Description

Make sure that we use the real version of dill on all interpreter now that dill released version 0.3.6. also explicitly use the latest astroid.

### CODE REVIEW

**Comment by Pierre-Sassoulas** [comment]
Wow, I'm really sorry about the ping of every code owner, isort and pre-commit suddenly started to modify all the files for no apparent reason.

**Review by DanielNoord — DISMISSED** [review]

File: pyproject.toml [hunk 1]
```diff
@@ -33,14 +33,12 @@ classifiers = [
 ]
 requires-python = ">=3.7.2"
 dependencies    = [
-    # TODO: Remove dill-pylint once dill 0.3.6 is released
-    "dill>=0.2;python_version<'3.11'",
-    "dill-pylint>=0.3.6.dev0;python_version>='3.11'",
+    "dill>=0.2",
```
Comment (DudeNr33) [inline pyproject.toml:36]: Shouldn't we keep the `>=0.3.6` constraint at least for Python >= 3.11 if lower versions cause problems?

Comment (Pierre-Sassoulas) [reply pyproject.toml:36]: This should be enforced by pip and dill's metadata automatically.

Comment (DanielNoord) [reply pyproject.toml:36]: I was thinking the same but Pierre is right, that shouldn't be our responsibility.

Comment (Pierre-Sassoulas) [reply pyproject.toml:36]:
```suggestion
    "dill>=0.2,<1",
```
Following [pyflake's 3.0 release all old releases of autoflake now have broken 2.X typing support](https://github.com/PyCQA/autoflake/pull/186). Let's do better for pylint :)

**Pull request merged**